src/Scrutor/IServiceTypeSelector.cs

```diff
@@ -41,6 +42,19 @@ public interface IServiceTypeSelector : IImplementationTypeSelector
         /// </summary>
         /// <param name="selector">A function to select service types based on implementation types.</param>
         /// <exception cref="ArgumentNullException">If the <paramref name="selector"/> argument is <c>null</c>.</exception>
-        ILifetimeSelector As(Func<Type, IEnumerable<Type>> selector);
+        ILifetimeSelector As(Func<Type, IEnumerable<Type>> selector);
+
+        /// <summary>
+        /// Registers the type with the first found matching interface name.  (e.g. ClassName is matched to IClassName)
+        /// </summary>
+        /// <returns></returns>
+        ILifetimeSelector WithMatchingInterface();

```

Comment:
Ideally, I'd like to stick to the `As*` naming here. Since this is pickinging the type you want to register *as* :smile: